### PR TITLE
Adding nitpick option howto to docs configuration

### PR DESCRIPTION
**Pull request by bsipocz**
One need to remove the comments to enable it, but I think it's useful to have in the template.

### CODE REVIEW

**Comment by bsipocz** [comment]
@astrofrog - Do you think if we can remove the python 2.6 build from the travis template? Maybe py3.3, too?

**Comment by bsipocz** [comment]
Well, #183 answered the question above, so this is rebased now.

**Comment by astrofrog** [comment]
Looks good to me - can you include an entry in the ``TEMPLATE_CHANGES`` file? Once we tag a new version of the package-template, we can advertise this on the affiliated package mailing list to encourage developers to turn this on.

File: docs/conf.py [hunk 1]
```diff
@@ -158,3 +158,16 @@
...
     edit_on_github_source_root = ""
     edit_on_github_doc_root = "docs"
+
+
+# -- Turn on nitpicky mode for sphinx (to warn about references not found) ----
+#
+# nitpicky = True
+# nitpick_ignore = []
+#
+# for line in open('nitpick-exceptions'):
```
Comment (astrofrog) [inline docs/conf.py]: Actually I think this should maybe first check if the file exists?

**Comment by bsipocz** [comment]
Do you think we can remove the comments and opt in to this by default?

**Comment by astrofrog** [comment]
@bsipocz - we could do actually. It might mean a bit more work when people next update to the latest template, but we can always tell them they can comment it out if they really don't have time to fix the (real) issues. If you remove the comments, maybe also include an empty exceptions file with some commented out entries for demonstration?

**Comment by astrofrog** [comment]
@bsipocz - can you rebase this? Also please don't include this yet by default - what I'd like to suggest is that in the next 'release' of the package template we tell people about this and encourage them to enable it, then in the following 'release' we enable it by default.

**Comment by astrofrog** [comment]
Note that we should put merging this on hold as the cookiecutter branch is now set up and we should no longer commit directly to master. Instead we should make this change in the cookiecutter branch. We still need to set up docs about how to use the cookiecutter branch, but once that's done it should be clear how to do this.

**Comment by astrofrog** [comment]
Closing in favour of https://github.com/astropy/package-template/pull/308 as I couldn't force push to this branch.

**Pull request closed without merging**